neuralmonkey/trainers/optimizers.py

```diff
@@ -0,0 +1,77 @@
+"""Collection of helper functions providing optimizers for trainers."""
```

Comment:
Jak už tady někde psal @cifkao,  tyhlety helpery vůbec nepotřebujeme, kontruktory optimizerů sde dají volat rovnou z konfiguráku. Ten learning rate decay be dal pak pořešit tím, že místo `learning_rate * lr_decay(global_step)`, které je v tomhle souboru třikrát by funkce, by decay vracel tensor, co by byl výsledek toho násobení. V případě konstrantního learning ratu, by se napsal do konfiguráku float.

Reply:
Souhlasím.

Možná by nebylo od věci to API udělat tak, aby se tam dalo použít např. [`tf.train.exponential_decay`](https://www.tensorflow.org/api_docs/python/tf/train/exponential_decay).

Reply:
yes. cokoliv co bude rozumnější. Už jsem předesílal, že to, jak je to udělaný teď, se mi vůbec nelíbí